kbd_backlight.cpp

```diff
@@ -269,21 +273,56 @@ void brightness_control(const std::string &brightnessPath,
   }
 }
 
-void read_events(int devFd, const std::string &brightnessPath) {
+void read_events(int devFd, const std::string &brightnessPath,
+				 const std::vector<int> &ignoredKeys, bool showPressedKeys) {
+  int ignoreNextValues = 0;
   while (!end_) {
 	struct input_event ie = {};
 	int rd = read(devFd, &ie, sizeof(struct input_event));
-	if (rd != 0) {
-	  lastEvent_ = std::chrono::system_clock::now();
+	if (rd != 0){
+		if (showPressedKeys && ie.type == 4 && ie.code == 4)
+		{
+			printf("Pressed key value: %d\n", ie.value);
+			fflush(stdout);
+		}
+		
+		bool correctKey = true;
+		if (ie.type == 4 && ie.code == 4) {
+			if (std::find(std::begin(ignoredKeys), 
+						  std::end(ignoredKeys), 
+						  ie.value) != std::end(ignoredKeys)) {
+				correctKey = false;
+				ignoreNextValues = 2;
```

Comment:
I guess ``ignoreNextValues`` is used to ignore the key up event (1).  Why is it set to 2?

Reply:
There are 3 events called after every key press, even when holding:
```
Processing key type: 4, code: 4, value: 200
Processing key type: 1, code: 103, value: 1
Processing key type: 0, code: 0, value: 0
```
And when holding key:
```
Processing key type: 4, code: 4, value: 200
Processing key type: 1, code: 103, value: 2
Processing key type: 0, code: 0, value: 0
```

Reply:
So, I catch the first scan event, and ignore the two more, that are generated.

Reply:
Good to know. Might be a good idea to add a comment in the code to document this.